packages/xrpl/src/sugar/submit.ts

```diff
@@ -119,29 +120,42 @@ async function submitRequest(
 async function waitForFinalTransactionOutcome(
   client: Client,
   txHash: string,
+  lastLedger: number,
 ): Promise<TxResponse> {
   await sleep(LEDGER_CLOSE_TIME)
 
-  const txResponse = await client.request({
-    command: 'tx',
-    transaction: txHash,
-  })
+  const latestLedger = await client.getLedgerIndex()
+
+  const txResponse = await client
+    .request({
+      command: 'tx',
+      transaction: txHash,
+    })
+    .catch(async (error) => {
+      // this type of error does not have its interface, type-assert and extract the value we need.
+      // eslint-disable-next-line @typescript-eslint/consistent-type-assertions,@typescript-eslint/no-unsafe-member-access -- L131
+      const message = error.data.error as string
+      if (message === 'txnNotFound') {
+        if (lastLedger > (await client.getLedgerIndex())) {
```

Comment:
This should be the other way around.
```ts
lastLedger < (await client.getLedgerIndex()
```

Reply:
We want to throw when the current index is greater than the lastLedgerSequence

Reply:
yes